tutorials/chapter1.md

```diff
@@ -58,7 +58,14 @@ There are two important elements in this class that are always required when pro
 * The ```publish``` method expects a const reference to the publishing object. All publisher implementations make a copy of this object before proceeding to the actual publication. There is another API for publication that skip the copy, that we will cover later.
 * Application code always use pointers to ```kpsr::Publisher```. This is done in order to keep the application independent of the implementation of the publisher. It is not only a good practice, but also necessary when building unit tests for classes like this example. The same principle applies to the ```kpsr::Subscriber``` and ```kpsr::Environment```.
 
-It is also important to note that the IDs of the different threads that are running during the execution of each example are printed out in order to clarify the number of threads involved in each example. Then, this is de purpose of the line where the thread ID is printed out in class SimplePublisher.
+It is also important to note that the IDs of the different threads that are running during the execution of each example are printed out in order to clarify the number of threads involved in each example. Then, along the next examples will run the next threads:
+
+* Main thread: thread running main(). Obviously, it runs in all examples. 
+* Eventloop Listener thread: thread running the EventLoop listener, which is the only thread that will handle all the listeners associated to all subscribers. It runs from example number 2.
+* Vector publisher thread: thread running the vector publisher. It runs from example number 3. 
+* DataMultiplexer Consumer 1 (Mod): thread running the module listener (note that it is independent of the other DataMultiplexer listener, Consumer 2).It runs at example 5.
+* DataMultiplexer Consumer 2 (Sum): thread running the sum listener (note that it is independent of the other DataMultiplexer listener, Consumer 1). It runs at example 5.
+
```

Comment:
It might be better to rearrange these explanations a little. At this stage in the tutorial the eventloop or datamultiplexer is not yet introduced. So better way would be to have here:

> We print the thread id in the publish and subscribe methods to demonstrate the differences between the different APIs and their threading models. Each example prints the id of the main thread (which is called from the `main()` function. 

Then when you have the example 2, you explain what is the listener thread. After the example 3 explanation, you write about the vector publisher thread. And finally you add the lines of the datamultiplexer after example 5.

Reply:
Done